admin/data/sql/migrations/20190306160234.do.fix-pupils-eligible-for-restart-view.sql

```diff
@@ -0,0 +1,44 @@
+ALTER VIEW [mtc_admin].[vewPupilsEligibleForRestart] AS
+
+  SELECT
+    p.id,
+    p.foreName,
+    p.middleNames,
+    p.lastName,
+    p.dateOfBirth,
+    p.urlSlug,
+    p.school_id,
+    count(*) as totalCheckCount
+  FROM
+    [mtc_admin].[pupil] p
+      LEFT JOIN [mtc_admin].[pupilAttendance] pa ON (p.id = pa.pupil_id and pa.isDeleted = 0)
+      INNER JOIN [mtc_admin].[check] AS chk ON (p.id = chk.pupil_id)
+      INNER JOIN [mtc_admin].[checkStatus] AS chkStatus ON (chk.checkStatus_id = chkStatus.id)
+  WHERE
+    -- don’t select pupils who are not attending
+    pa.id IS NULL
+    -- pupils must have already attempted 1 or more checks that are started, complete, not received
+    AND   chkStatus.code IN ('STD', 'CMP', 'NTR')
+    AND   chk.isLiveCheck = 1
+    AND   p.id NOT IN (
+      -- remove pupils with unconsumed restarts
+      SELECT
+        p.id
```

Comment:
@jon-shipley shouldn't that be p2.id?
I was also wondering if checking the pupil status instead would be helpful and potentially provide a simpler solution?

Reply:
Updated to use p2.id - good catch 👍 

I looked at using pupil status as well, but this seems impossible as the pupil status does not directly encode restart status.